minetest/minetest
# [no squash] Avoid use of select(2)

`select()` is a [fundamentally broken API](https://stackoverflow.com/questions/12818634/giving-a-socket-descriptor-value-greater-than-1024-in-select) that shouldn't be used anymore. So this is what we'll do.

This raises the minimum curl version to at least 7.28.0. It was released sometime in 2012.

## To do

This PR is Ready for Review.

## How to test

1. try single player
2. try mainmenu
3. try this nearly pointless mod that performs HTTP requests:
```lua
local http_api = core.request_http_api()
assert(http_api)

local testn = 0
local function dotest()
	if testn >= 6 then
		print("done")
		return
	end
	testn = testn + 1
	local my_n = testn
	print("begin " .. my_n)
	http_api.fetch({
		url = "https://httpbin.org/delay/2",
	}, function(res)
		print("end " .. my_n .. " " .. tostring(res.succeeded))
	end)
	core.after(1 + testn/4, dotest)
end
core.after(3, dotest)
```
4. ensure it doesn't hang at shutdown
